### PR TITLE
refactor: remove superfluous boost::thread_interrupted catch

### PR DESCRIPTION
**Pull request by hmel**
No point in catching exception just to re-throw it

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Please don't mark non-doc code changes trivial.

**Comment by fanquake** [comment]
@hmel You can use `refactor:`. Please squash the commits and update commit message.

File: src/bitcoin-cli.cpp [hunk 1]
```diff
@@ -494,9 +494,6 @@ static int CommandLineRPC(int argc, char *argv[])
             }
         } while (fWait);
     }
-    catch (const boost::thread_interrupted&) {
-        throw;
-    }
     catch (const std::exception& e) {
```
Comment (Empact) [inline src/bitcoin-cli.cpp]: The catch above is to give special treatment to `boost::thread_interrupted`, relative to the `std::exception` catch that follows. It's a way of excluding these from the `std::exception` treatment.

**Comment by Empact** [comment]
NACK 1757d88

**Pull request closed without merging**